k8s-infra.yml

```diff
@@ -0,0 +1,37 @@
+---
+#####
+## Playbook that provisions and then configures a Kubernetes
+## cluster on OpenStack
+#####
+
+# Setting this in a separate play is the only way to avoid a recursive templating error
+- import_playbook: k8s-infra-variables.yml
+  when: cluster_state | lower == 'present'
+
+# Provision the cluster infrastructure
+- hosts: openstack
+  roles:
+    - role: jasmin.cluster-infra
+      cluster_groups:
+        - "{{ cluster_gw }}"
+        - "{{ cluster_masters }}"
+        - "{{ cluster_workers }}"
+      when: cluster_state | lower == 'present'
+
+# Do the configuration
+- import_playbook: k8s-infra-configure.yml
+  vars:
+    cloud_provider: openstack
+    openstack_trust_id: "{{ hostvars['localhost']['openstack_trust_id'] }}"
```

Comment:
Instead of `localhost`, use `groups['openstack'] | first`. It is not guaranteed that the OpenStack machine is the local machine, or that it is called `localhost` in the inventory.